javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrintVisitor.java

```diff
@@ -1380,8 +1380,23 @@ public void visit(TypeExpr n, Void arg) {
 
     @Override
     public void visit(NodeList n, Void arg) {
-        for (Object node : n) {
-            ((Node) node).accept(this, arg);
+        if (configuration.isOrderImports() && n.size() > 0 && n.get(0) instanceof ImportDeclaration) {
+            //noinspection unchecked
+            NodeList<ImportDeclaration> modifiableList = new NodeList<>(n);
+            modifiableList.sort((left, right) -> {
+                int sort = Integer.compare(left.isStatic() ? 0 : 1, right.isStatic() ? 0 : 1);
+                if (sort == 0) {
+                    sort = left.getNameAsString().compareTo(right.getNameAsString());
+                }
+                return sort;
+            });
+            for (Object node : modifiableList) {
+                ((Node) node).accept(this, arg);
+            }
+        } else {
```

Comment:
The import stuff could be extracted into a separate method so that the common case doesn't get hidden away at the end.

Reply:
I'm still struggling with this, since we don't want to mutate the original list. Perhaps something like
``` java
public void visit(NodeList n, Void arg) {
    NodeList<ImportDeclaration> sortedList = processOrderOfImports(n);
    for (Object node : sortedList) {
        ((Node) node).accept(this, arg);
    }
}
```
...and move the sort logic to `processOrderOfImports()`? The problem is that this method is used everywhere. Is it possible to differentiate a list of imports before getting here?

Reply:
Hmmm, I see. Leave it like this then :-)